### DISCUSSION TITLE
.npmrc legacy-peer-deps ignored

### DISCUSSION THREAD

**Discussion by yoavain** [renovatebot/renovate#16613] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Let's say a dependency has limited its peer dependency to a specific version. 
Typically, this would block you from updating this peer dependency (using npm v7 or higher).
What if this dependency can work just fine with the new version of this peer dependency? This might be a cause of a package not being maintained frequently (or at all).

npm gives 2 ways to ignore this conflict:
1. using cli with `--force` or `--legacy-peer-deps`
2. using .npmrc file with `legacy-peer-deps=true`

Assuming, you want to make this persistent across all environments (between developers, ci/cd), you would probably choose the latter. Yes, this is not good practice, this will result in ignoring other conflicts, but this is also a valid approach when migrating from NodeJS 14 to NodeJS 16 since it keeps the old peer dependencies resolution behavior..

Renovate ignores the `legacy-peer-deps=true` in the .npmrc file.
It results in an "Artifact update problem" error.
Failing to update `package-lock.json` file and opening a PR with only package.json file

A small reproducible:
Node 16.6.0
Npm 8.11

package.json
```json
{
  "name": "renovate-repro",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-fittext": "1.0.0"
  }
}
```
.npmrc
```
save-exact=true
package-lock=true
legacy-peer-deps=true
```

Renovate tries to update react to 18.2, resulting in a bad PR.
Running `npm install` works just fine because of the `legacy-peer-deps=true` in the .npmrc file. Without it - it fails too.

I also have a test repository where this happens:
https://github.com/yoavain/renovate-playground/pull/107

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by rarkins** [comment, marked as answer]
The behavior you're seeing is because you added this setting: https://github.com/yoavain/renovate-playground/blob/9e7a1e8159a1e392a65b94d0eed35fc7f5125db2/.github/renovate.json#L31

**Reply by yoavain** [reply]
Thanks, missed that
